### ISSUE TITLE
Uncaught exceptions seen when enabling UPnP in some configurations

### ISSUE THREAD

**Issue by soda3x** [soda3x/ArmaReforgerServerTool#106]
Below is a users logs from the uncaught `SocketException`:

```txt
See the end of this message for details on invoking 
just-in-time (JIT) debugging instead of this dialog box.

************** Exception Text **************
System.Net.Sockets.SocketException (10054): An existing connection was forcibly closed by the remote host.
   at System.Net.Sockets.Socket.ReceiveFrom(Byte[] buffer, Int32 offset, Int32 size, SocketFlags socketFlags, EndPoint& remoteEP)
   at System.Net.Sockets.UdpClient.Receive(IPEndPoint& remoteEP)
   at Open.Nat.Searcher.Receive(CancellationToken cancelationToken)
   at Open.Nat.Searcher.<>c__DisplayClass4_0.<Search>b__0(Object _)
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.Tasks.Task.<>c.<.cctor>b__272_0(Object obj)
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location ---
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
--- End of stack trace from previous location ---
   at Open.Nat.Searcher.Search(CancellationToken cancelationToken)
   at Open.Nat.NatDiscoverer.DiscoverAsync(PortMapper portMapper, Boolean onlyOne, CancellationTokenSource cts)
   at Open.Nat.NatDiscoverer.DiscoverDeviceAsync(PortMapper portMapper, CancellationTokenSource cancellationTokenSource)
   at Open.Nat.NatDiscoverer.DiscoverDeviceAsync()
   at ReforgerServerApp.Managers.NetworkManager.ConfigurePortMappings(List`1 mappings)
   at ReforgerServerApp.Managers.ProcessManager.StartStopServer()
   at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__128_0(Object state)


************** Loaded Assemblies **************
System.Private.CoreLib
    Assembly Version: 6.0.0.0
    Win32 Version: 6.0.2824.12007
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/6.0.28/System.Private.CoreLib.dll
----------------------------------------
ReforgerServerApp
    Assembly Version: 0.9.2.0
    Win32 Version: 0.9.2
    CodeBase: file:///C:/Users/Administrator/AppData/Local/ArmaReforgerServerTool/ReforgerServerApp.dll
----------------------------------------
System.Runtime
    Assembly Version: 6.0.0.0
    Win32 Version: 6.0.2824.12007
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/6.0.28/System.Runtime.dll
----------------------------------------
Serilog
    Assembly Version: 4.0.0.0
    Win32 Version: 4.0.0.0
    CodeBase: file:///C:/Users/Administrator/AppData/Local/ArmaReforgerServerTool/Serilog.dll
----------------------------------------
Serilog.Sinks.Console
    Assembly Version: 6.0.0.0
    Win32 Version: 6.0.0.0
    CodeBase: file:///C:/Users/Administrator/AppData/Local/ArmaReforgerServerTool/Serilog.Sinks.Console.dll
----------------------------------------
Serilog.Sinks.File
    Assembly Version: 6.0.0.0
    Win32 Version: 6.0.0.0
    CodeBase: file:///C:/Users/Administrator/AppData/Local/ArmaReforgerServerTool/Serilog.Sinks.File.dll
----------------------------------------
System.Windows.Forms
    Assembly Version: 6.0.2.0
    Win32 Version: 6.0.2824.12004
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.WindowsDesktop.App/6.0.28/System.Windows.Forms.dll
----------------------------------------
System.ComponentModel.Primitives
    Assembly Version: 6.0.0.0
    Win32 Version: 6.0.2824.12007
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/6.0.28/System.ComponentModel.Primitives.dll
----------------------------------------
System.Windows.Forms.Primitives
    Assembly Version: 6.0.2.0
    Win32 Version: 6.0.2824.12004
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.WindowsDesktop.App/6.0.28/System.Windows.Forms.Primitives.dll
----------------------------------------
System.Runtime.InteropServices
    Assembly Version: 6.0.0.0
    Win32 Version: 6.0.2824.12007
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/6.0.28/System.Runtime.InteropServices.dll
----------------------------------------
System.Drawing.Primitives
    Assembly Version: 6.0.0.0
    Win32 Version: 6.0.2824.12007
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/6.0.28/System.Drawing.Primitives.dll
----------------------------------------
System.Collections.Specialized
    Assembly Version: 6.0.0.0
    Win32 Version: 6.0.2824.12007
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/6.0.28/System.Collections.Specialized.dll
----------------------------------------
System.Text.Json
    Assembly Version: 6.0.0.0
    Win32 Version: 6.0.2824.12007
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/6.0.28/System.Text.Json.dll
----------------------------------------
System.Collections
    Assembly Version: 6.0.0.0
    Win32 Version: 6.0.2824.12007
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/6.0.28/System.Collections.dll
----------------------------------------
System.Collections.Concurrent
    Assembly Version: 6.0.0.0
    Win32 Version: 6.0.2824.12007
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/6.0.28/System.Collections.Concurrent.dll
... (386 lines left)
```

**Comment by soda3x** [comment]
Also seems to sometimes through a `NatDeviceNotFoundException`:

```txt
See the end of this message for details on invoking 
just-in-time (JIT) debugging instead of this dialog box.

************** Exception Text **************
Open.Nat.NatDeviceNotFoundException: Exception of type 'Open.Nat.NatDeviceNotFoundException' was thrown.
   at Open.Nat.NatDiscoverer.DiscoverDeviceAsync(PortMapper portMapper, CancellationTokenSource cancellationTokenSource)
   at Open.Nat.NatDiscoverer.DiscoverDeviceAsync()
   at ReforgerServerApp.Managers.NetworkManager.ConfigurePortMappings(List`1 mappings)
   at ReforgerServerApp.Managers.ProcessManager.StartStopServer()
   at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__128_0(Object state)


************** Loaded Assemblies **************
System.Private.CoreLib
    Assembly Version: 6.0.0.0
    Win32 Version: 6.0.3624.51421
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/6.0.36/System.Private.CoreLib.dll
----------------------------------------
ReforgerServerApp
    Assembly Version: 0.9.2.0
    Win32 Version: 0.9.2
    CodeBase: file:///C:/Users/joshs/AppData/Local/ArmaReforgerServerTool/ReforgerServerApp.dll
----------------------------------------
System.Runtime
    Assembly Version: 6.0.0.0
    Win32 Version: 6.0.3624.51421
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/6.0.36/System.Runtime.dll
----------------------------------------
Serilog
    Assembly Version: 4.0.0.0
    Win32 Version: 4.0.0.0
    CodeBase: file:///C:/Users/joshs/AppData/Local/ArmaReforgerServerTool/Serilog.dll
----------------------------------------
Serilog.Sinks.Console
    Assembly Version: 6.0.0.0
    Win32 Version: 6.0.0.0
    CodeBase: file:///C:/Users/joshs/AppData/Local/ArmaReforgerServerTool/Serilog.Sinks.Console.dll
----------------------------------------
Serilog.Sinks.File
    Assembly Version: 6.0.0.0
    Win32 Version: 6.0.0.0
    CodeBase: file:///C:/Users/joshs/AppData/Local/ArmaReforgerServerTool/Serilog.Sinks.File.dll
----------------------------------------
System.Windows.Forms
    Assembly Version: 6.0.2.0
    Win32 Version: 6.0.3624.51513
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.WindowsDesktop.App/6.0.36/System.Windows.Forms.dll
----------------------------------------
System.ComponentModel.Primitives
    Assembly Version: 6.0.0.0
    Win32 Version: 6.0.3624.51421
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/6.0.36/System.ComponentModel.Primitives.dll
----------------------------------------
System.Windows.Forms.Primitives
    Assembly Version: 6.0.2.0
    Win32 Version: 6.0.3624.51513
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.WindowsDesktop.App/6.0.36/System.Windows.Forms.Primitives.dll
----------------------------------------
System.Runtime.InteropServices
    Assembly Version: 6.0.0.0
    Win32 Version: 6.0.3624.51421
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/6.0.36/System.Runtime.InteropServices.dll
----------------------------------------
System.Drawing.Primitives
    Assembly Version: 6.0.0.0
    Win32 Version: 6.0.3624.51421
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/6.0.36/System.Drawing.Primitives.dll
----------------------------------------
System.Collections.Specialized
    Assembly Version: 6.0.0.0
    Win32 Version: 6.0.3624.51421
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/6.0.36/System.Collections.Specialized.dll
----------------------------------------
System.Text.Json
    Assembly Version: 6.0.0.0
    Win32 Version: 6.0.3624.51421
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/6.0.36/System.Text.Json.dll
----------------------------------------
System.Collections
    Assembly Version: 6.0.0.0
    Win32 Version: 6.0.3624.51421
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/6.0.36/System.Collections.dll
----------------------------------------
System.Collections.Concurrent
    Assembly Version: 6.0.0.0
    Win32 Version: 6.0.3624.51421
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/6.0.36/System.Collections.Concurrent.dll
----------------------------------------
System.Private.Uri
    Assembly Version: 6.0.0.0
    Win32 Version: 6.0.3624.51421
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/6.0.36/System.Private.Uri.dll
----------------------------------------
System.Reflection.Emit.Lightweight
    Assembly Version: 6.0.0.0
    Win32 Version: 6.0.3624.51421
    CodeBase: file:///C:/Program%20Files/dotnet/shared/Microsoft.NETCore.App/6.0.36/System.Reflection.Emit.Lightweight.dll
----------------------------------------
System.Reflection.Primitives
    Assembly Version: 6.0.0.0
... (358 lines left)
```

**Comment by soda3x** [comment]
Fixed in `0.9.2.1`
